paprika_recipes/remote.py

```diff
@@ -75,6 +76,30 @@ def get_recipe_by_id(self, id: str, hash: str) -> RemoteRecipe:
     def count(self) -> int:
         return len(self._get_remote_recipe_identifiers())
 
+    def download_image(self, recipe: RemoteRecipe):
+
+        if recipe.image_url:
+            result = requests.request('GET', recipe.image_url)
+            result.raise_for_status()
+
+            if not os.path.exists('images'):
+                os.mkdir('images')
+
+            url = recipe.image_url
+
+            # Remove query parameters from the URL.
+            if url.find('?') != -1:
```

Comment:
I think what you're looking for here is `urllib.parse.urlparse` & `os.path.basename` e.g:

```python
filename = os.path.basename(urllib.parse.urlparse(url).path)
```

& if you're unfamiliar:

```
>>> urllib.parse.urlparse('https://google.com/somewhere.jpg?test=1')
ParseResult(scheme='https', netloc='google.com', path='/somewhere.jpg', params='', query='test=1', fragment='')
>>> import os
>>> os.path.basename('/somewhere.jpg')
'somewhere.jpg'
```